pycobertura/reporters.py

```diff
@@ -274,6 +287,26 @@ def generate(self):
         return colored_json_string
 
 
+class YamlReporterDelta(DeltaReporter):
+    not_available = None
+
+    def generate(self):
+        lines = self.get_report_lines()
+
+        if self.show_source:
+            missed_lines_colored = [
+                self.color_number(line) for line in lines["Missing"]
+            ]
+            lines["Missing"] = missed_lines_colored
+
+        rows = {k: v[:-1] for k, v in lines.items()}
+        footer = {k: v[-1] for k, v in lines.items() if k != "Missing"}
+        # need to write to a buffer as yml packages are using a streaming interface
+        buf = io.BytesIO()
+        yaml.YAML().dump({"total": footer, "files": rows}, buf)
+        return buf.getvalue().replace(rb"\e", b"\x1b")
```

Comment:
Can we leave a comment here about why we're doing this `.replace()`?

Reply:
Kind of difficult I just realized that ANSI color code to work needs \x1b and for some reason the YAML dump replaces that with \e and then it does not work anymore, which is why I back-convert it, but unfortunately it is a bit of a hack. So I am not quite sure on how to comment on this :)

Reply:
Honestly, I wouldn't worry about the color for formats that are meant to be machine-readable, such as JSON, CSV, YAML. Colors are best for humans. What do you think?

But a quick Google search tells me that `\e` and `\x1b` are both escape characters.

https://stackoverflow.com/questions/15011478/ansi-questions-x1b25h-and-x1be
https://stackoverflow.com/questions/26153308/best-ansi-escape-beginning

Reply:
The problem is that in Visual Studio Code on my machine it only worked with '\x1b`, i.e. the output was coloured, whereas with `\e` the output did not show as coloured but instead show the characters non-interpreted, i.e. with `\e` in there...

Reply:
It is also in the official Python docs: https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals

Reply:
busted added a comment in https://github.com/aconrad/pycobertura/pull/131/commits/149af62dc26cc3ac1c8e5a909912a50cb0a8ca70